File: java/src/main/java/mx/iteso/adapter/googledoc/IGoogleDoc.java
package mx.iteso.adapter.googledoc;

import mx.iteso.adapter.Font;
import mx.iteso.adapter.Image;
import mx.iteso.adapter.Style;

/**
 * IGoogleDoc Interface.
 */
public interface IGoogleDoc {
    /**
     * Font getter.
     * @return GoogleDoc's Font.
     */
    Font getFont();
    /**
     * Style getter.
     * @return GoogleDoc's style.
     */
    Style getStyle();
    /**
     * Background image getter.
     * @return GoogleDoc's background image.
     */
    Image getBackground();
    /**
     * Sharing permissions setter.
     * @param sharingPermissions Sharing permissions.
     */
    void setSharingPermissions(int sharingPermissions);
}

File: csharp/adapter_practice/main/ints/IGoogleDoc.cs
using adapter_practice.main.other;
using System;
using System.Collections.Generic;
using System.Text;

namespace adapter_practice.main.ints
{
    public interface IGoogleDoc
    {
        /**
         * Gets Font.
         * @return font.
         */
        Font getFont();
        /**
         * @return object of style.
         */
        Object getStyle();
        /**
         * @return Background Image of document.
         */
        BackgroundImage getBackground();
        /**
         *
         * @param sharingPermissions sets permissions.
         */
        void setSharingPermissions(int sharingPermissions);
    }
}
